### PR TITLE
#2183 granular permissions

### LINKED ISSUE #2299: add dictionary of enum -> object/bool to `UserDto` & `User`
Should sit alongside the access property until the front end is changed to use the new dictionary

Value will need to be an object so the contract isn't broken in future, the object could simply be an empty dto for now.

e.g. (for cosmos)

```json
{
  "emailAddress": "simon@email.com",
  "permissions": { // the new dictionary 
    "manageAccess": {},
    // unpermitted access is omitted
  },
  "access": { // the existing Access object
    "manageAccess": true,
    "manageGames": false,
  }
}
```

The front end _could_ use bool for the RHS rather than an object and use truthiness to compute access, however maybe it makes sense to describe the final format initially rather than changing it

### PR DESCRIPTION
**Pull request by laingsimon**
Resolves #2299

### CODE REVIEW

**Comment by laingsimon** [comment]
### Build and test output









#### Dotnet tests:
```
Test run for /home/runner/work/courage_scores/courage_scores/CourageScores.StubCosmos.Tests/bin/Release/net10.0/CourageScores.StubCosmos.Tests.dll (.NETCoreApp,Version=v10.0)
Test run for /home/runner/work/courage_scores/courage_scores/CourageScores.Tests/bin/Release/net10.0/CourageScores.Tests.dll (.NETCoreApp,Version=v10.0)
A total of 1 test files matched the specified pattern.A total of 1 test files matched the specified pattern.


Passed!  - Failed:     0, Passed:    72, Skipped:     0, Total:    72, Duration: 718 ms - CourageScores.StubCosmos.Tests.dll (net10.0)
```
#### React tests:
```
  Test Suites: 139 passed, 139 total
  Tests:       1 skipped, 2196 passed, 2197 total
  Snapshots:   0 total
  Time:        31.283 s
  Ran all test suites.
```
#### Playwright tests:
```
Running 2 tests using 1 worker
··
  2 passed (50.5s)
```

[Dotnet Logs](https://github.com/laingsimon/courage_scores/actions/runs/28299972317/job/83846487556?pr=2315) | [React Logs](https://github.com/laingsimon/courage_scores/actions/runs/28299972317/job/83846487580?pr=2315) | [Playwright logs](https://github.com/laingsimon/courage_scores/actions/runs/28299972317/job/83846487570?pr=2315) | [Analysis](https://github.com/laingsimon/courage_scores/actions/runs/28300034048/job/83846633686)

File: CourageScores/Services/Identity/AccessService.cs [hunk 1]
```diff
@@ -11,7 +12,14 @@ public Task<bool> HasAccess(UserDto? user, AccessOption access, CancellationToke
         return Task.FromResult(GetAccess(userAccess, access));
     }
 
-    private static bool GetAccess(AccessDto userAccess, AccessOption access)
+    public Task<bool> HasAccess(User? user, AccessOption access, CancellationToken token)
```
Comment (laingsimon) [inline CourageScores/Services/Identity/AccessService.cs:15]: Add tests

File: CourageScores.Tests/Models/Adapters/Identity/AccessLevelAdapterTests.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    [SetUp]
+    public void SetupEachTest()
+    {
+        _accessAdapter = new MockSimpleAdapter<Access, AccessDto>(Access, AccessDto);
+        _adapter = new AccessLevelAdapter(_accessAdapter);
+    }
+
+    [Test]
+    public async Task AddAccess_GivenAUserAccessUpdate_AdaptsAccessAndAccessLevels()
```
Comment (laingsimon) [inline CourageScores.Tests/Models/Adapters/Identity/AccessLevelAdapterTests.cs:48]: Add tests for when the access is null
